server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/idp/NonSnarlIdpMetadataManager.java

```diff
@@ -213,7 +213,7 @@ protected String getSpName(ExtendedMetadataDelegate provider) throws MetadataPro
     protected String getHostedSpName(ExtendedMetadataDelegate provider) throws MetadataProviderException {
         String key = getSpName(provider);
         ExtendedMetadata extendedMetadata = getExtendedMetadata(key, provider);
-        if (extendedMetadata.isLocal()) {
+        if (extendedMetadata != null && extendedMetadata.isLocal()) {
```

Comment:
If extendedMetadata is null, it might be better to throw MetadataProviderException instead of returning null.